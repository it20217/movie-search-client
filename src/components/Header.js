import logo from "../assets/images/dev-logo.png"

function Header() {

  return(
    <div>
      <img src={logo} width={"100"} alt="Logo" />
      <br/>
    </div>
  )
}

export default Header;