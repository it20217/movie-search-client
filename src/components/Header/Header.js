import { useState, useEffect, useCallback } from 'react'
import { useNavigate, useLocation } from "react-router-dom"
import logo from "../../assets/images/dev-logo.png"
import SearchForm  from '../SearchForm/SearchForm'
import styles from "./Header.module.scss";
import { HiOutlineMenu } from "react-icons/hi";

function Header(props) {

  const [query, setQuery] = useState('');
  const [path, setPath] = useState(false);
  const navigate = useNavigate();
  const location = useLocation();

  const handleInput = (e) => {
    e.preventDefault();
    setQuery(e.target.value);
  }
  const handleFind = (e) => {
    navigate(`/search/${query}`);
    setQuery('');
  }
  

 useEffect(() => {
    location.pathname.localeCompare("/") === 0 
  ?
    setPath(true)
  :
    setPath(false)

  }, [location]);

  return(
    <div className={styles.header}>
     <div 
      className={styles["app-logo"]}
      onClick={(e) => {e.preventDefault(); navigate('/')}}
     >
      <img src={logo}  alt="Logo" />
     </div>
      <div className={`${styles["app-name"]} text-gray-800`}>
        Movie app!
      </div>
      {!path && 

        <SearchForm />
       
      }
      
      <div 
        className={styles["menu-icon-div"]}
        onClick={props.setShowNavbar}
      >
        < HiOutlineMenu/>
      </div>
    </div>
    
  )
}

export default Header;

