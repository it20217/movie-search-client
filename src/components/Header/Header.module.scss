.header {
  display: flex;
  flex-direction: row;
  height: 100px;
  width: 100%;
  line-height: 100px;
  border-bottom: 2px solid #c9c9c9;
}
.app-name {
  font-size: 32px;
  font-weight: 600;
}