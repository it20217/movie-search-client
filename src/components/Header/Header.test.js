import React from 'react';
/** Snapshots rendering */
import renderer from "react-test-renderer";
/** Component */
import Header from './Header';
/** React router */
import { BrowserRouter as Router } from 'react-router-dom';

describe("Header page component", () => {
  it("renders correctly", () => {
    const component = renderer.create(<Router><Header/></Router>).toJSON();
    expect(component).toMatchSnapshot();
  })
})