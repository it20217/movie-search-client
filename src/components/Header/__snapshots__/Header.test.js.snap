// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header page component renders correctly 1`] = `
<div
  className="header"
>
  <div
    className="app-logo"
    onClick={[Function]}
  >
    <img
      alt="Logo"
      src="cinema-154392__340.webp"
    />
  </div>
  <div
    className="app-name text-gray-800"
  >
    Movie app!
  </div>
  <div
    className="search"
  >
    <div
      className="form-wrapper"
      style={
        Object {
          "marginTop": undefined,
        }
      }
    >
      <div
        className="search-input"
      >
        <input
          className="search-input"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Spiderman: No Way Home"
          style={
            Object {
              "borderColor": "#273461",
            }
          }
          type="search"
          value=""
        />
      </div>
      <div
        className="btn-container"
      >
        <button
          className="form-btn"
          onClick={[Function]}
          style={
            Object {
              "Color": "#273461",
              "borderColor": "#273461",
            }
          }
          type="button"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 5l7 7m0 0l-7 7m7-7H3"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <div
    className="menu-icon-div"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 6h16M4 12h16M4 18h16"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </div>
</div>
`;
